src/faebryk/library/SWDConnector.py

```diff
@@ -9,12 +9,26 @@
 class SWDConnector(Module):
     swd: F.SWD
     gnd_detect: F.ElectricLogic
-    vcc: F.ElectricPower
+    reference: F.ElectricPower
 
-    designator_prefix = L.f_field(F.has_designator_prefix_defined)("J")
+    designator_prefix = L.f_field(F.has_designator_prefix_defined)(
+        F.has_designator_prefix.Prefix.J
+    )
 
     @L.rt_field
     def single_electric_reference(self):
         return F.has_single_electric_reference_defined(
             F.ElectricLogic.connect_all_module_references(self)
         )
+
+    def __preinit__(self):
+        # ------------------------------------
+        #           connections
+        # ------------------------------------
+        self.get_trait(F.has_single_electric_reference).get_reference().connect(
+            self.reference
```

Comment:
That's not needed with single ref trait right? If it is, we should make it not be needed.